src/main/java/de/tum/in/www1/artemis/service/FeedbackService.java

```diff
@@ -94,6 +97,59 @@ public FeedbackService(FeedbackRepository feedbackRepository) {
         return feedbackList;
     }
 
+    /**
+     * Create an automatic feedback object from a test job.
+     * @param testJob the test case to be transformed.
+     * @param successful if the test case was successful.
+     * @param programmingLanguage the programming language of the exercise.
+     * @return Feedback object for the test job
+     */
+    public Feedback createFeedbackFromTestJob(BambooBuildResultNotificationDTO.BambooTestJobDTO testJob, boolean successful, final ProgrammingLanguage programmingLanguage) {
```

Comment:
Important: FeedbackService (and all other services) must be independent of Bamboo specifics. It should only depend on the more general `ContinuousIntegrationService`. This is an important design principle in Artemis to avoid issues with other VC / CI systems (Gitlab, Jenkins) and future implementations of those.

I think you have two options:
1) Move this method into Bamboo service
2) Find an abstraction for BambooTestJobDTO that can be used for Jenkins, Bamboo and potentially all other CI Services

The 2nd option is used for the StaticCodeAnalysisReportDTO which is basically independent from the concrete CI system. However, this does not mean that you also need to choose it here

Reply:
Ok good point. I think I'm just using the name and errors property of the BambooTestJobDTO. I could also just give these directly as parameters to the method.

Reply:
I updated the signature of the method which is now independent of bamboo-specific models.
Therefore it is now also used in the JenkinsService.